lib/notifier/index.js

```diff
@@ -0,0 +1,224 @@
+        // variables for setAlertFingerprint: (fingerprint = null, lastAlertTime = null)
+        utils.setAlertFingerprint(null, null)
+        if (inAlert) return utils.sendNoAlerts(plugins, smsEnabled, emailEnabled)
+      }
+      if (utils.shouldNotAlert(currentAlertFingerprint)) return
+
+      const message = buildMessage(alerts, notifications)
+      utils.setAlertFingerprint(currentAlertFingerprint, Date.now())
+      return plugins.sendMessage(message)
+    })
+    .then(results => {
+      if (results && results.length > 0) {
+        logger.debug('Successfully sent alerts')
+      }
+    })
+    .catch(logger.error)
+}
+
+function getAlerts (plugins) {
+  return Promise.all([
+    plugins.checkBalances(),
+    queries.machineEvents(),
+    plugins.getMachineNames()
+  ]).then(([balances, events, devices]) => {
+    notifyIfActive('balance', 'balancesNotify', balances).catch(console.error)
+    return buildAlerts(checkPings(devices), balances, events, devices)
+  })
+}
+
+function buildAlerts (pings, balances, events, devices) {
+  const alerts = { devices: {}, deviceNames: {} }
+  alerts.general = _.filter(r => !r.deviceId, balances)
+  _.forEach(device => {
+    const deviceId = device.deviceId
+    const deviceName = device.name
+    const deviceEvents = events.filter(function (eventRow) {
+      return eventRow.device_id === deviceId
+    })
+    const ping = pings[deviceId] || []
+    const stuckScreen = checkStuckScreen(deviceEvents, deviceName)
+
+    alerts.devices = _.set([deviceId, 'balanceAlerts'], _.filter(
+      ['deviceId', deviceId],
+      balances
+    ), alerts.devices)
+    alerts.devices[deviceId].deviceAlerts = _.isEmpty(ping) ? stuckScreen : ping
+
+    alerts.deviceNames[deviceId] = deviceName
+  }, devices)
+
+  return alerts
+}
+
+function checkPings (devices) {
+  const deviceIds = _.map('deviceId', devices)
+  const pings = _.map(utils.checkPing, devices)
+  return _.zipObject(deviceIds)(pings)
+}
+
+function checkStuckScreen (deviceEvents, machineName) {
+  const sortedEvents = _.sortBy(
+    utils.getDeviceTime,
+    _.map(utils.parseEventNote, deviceEvents)
+  )
+  const lastEvent = _.last(sortedEvents)
+
+  if (!lastEvent) return []
+
+  const state = lastEvent.note.state
+  const isIdle = lastEvent.note.isIdle
+
+  if (isIdle) return []
+
+  const age = Math.floor(lastEvent.age)
+  if (age > STALE_STATE) return [{ code: STALE, state, age, machineName }]
+
+  return []
+}
+
+function transactionNotify (tx, rec) {
+  return settingsLoader.loadLatest().then(settings => {
+    const notifSettings = configManager.getGlobalNotifications(settings.config)
+    const highValueTx = tx.fiat.gt(notifSettings.highValueTransaction || Infinity)
+    const isCashOut = tx.direction === 'cashOut'
+
+    // for notification center
+    const directionDisplay = tx.direction === 'cashOut' ? 'cash-out' : 'cash-in'
+    const readyToNotify = tx.direction === 'cashIn' || (tx.direction === 'cashOut' && rec.isRedemption)
```

Comment:
Why not reuse the variable `isCashOut`?
```suggestion
    const isCashOut = tx.direction === 'cashOut'

    // for notification center
    const directionDisplay = isCashOut ? 'cash-out' : 'cash-in'
    const readyToNotify = !isCashOut || (isCashOut && rec.isRedemption)
```